libs/wingsdk/src/target-awscdk/bucket.ts

```diff
@@ -0,0 +1,108 @@
+  private readonly bucket: S3Bucket;
+  private readonly public: boolean;
+
+  constructor(scope: Construct, id: string, props: cloud.BucketProps = {}) {
+    super(scope, id, props);
+
+    this.public = props.public ?? false;
+
+    this.bucket = new S3Bucket(this, "Default", {
+      encryption: BucketEncryption.S3_MANAGED,
+      blockPublicAccess: this.public ? undefined : BlockPublicAccess.BLOCK_ALL,
+      publicReadAccess: this.public ? true : false,
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+  }
+
+  public addObject(key: string, body: string): void {
+    new BucketDeployment(this, `S3Object-${key}`, {
+      destinationBucket: this.bucket,
+      sources: [Source.data(key, body)],
+    });
+  }
+
+  /** @internal */
+  public _bind(host: core.IInflightHost, ops: string[]): void {
+    if (!(host instanceof Function)) {
+      throw new Error("buckets can only be bound by tfaws.Function for now");
+    }
+
+    if (ops.includes(cloud.BucketInflightMethods.PUT)) {
+      host.addPolicyStatements({
+        effect: Effect.ALLOW,
+        actions: ["s3:PutObject*", "s3:Abort*"],
+        resources: [`${this.bucket.bucketArn}`, `${this.bucket.bucketArn}/*`],
+      });
+    }
+    if (ops.includes(cloud.BucketInflightMethods.GET)) {
+      host.addPolicyStatements({
+        effect: Effect.ALLOW,
+        actions: ["s3:GetObject*", "s3:GetBucket*", "s3:List*"],
+        resources: [`${this.bucket.bucketArn}`, `${this.bucket.bucketArn}/*`],
+      });
+    }
+    if (ops.includes(cloud.BucketInflightMethods.LIST)) {
+      host.addPolicyStatements({
+        effect: Effect.ALLOW,
+        actions: ["s3:GetObject*", "s3:GetBucket*", "s3:List*"],
+        resources: [`${this.bucket.bucketArn}`, `${this.bucket.bucketArn}/*`],
+      });
+    }
+    if (ops.includes(cloud.BucketInflightMethods.DELETE)) {
+      host.addPolicyStatements({
+        effect: Effect.ALLOW,
+        actions: [
+          "s3:DeleteObject*",
+          "s3:DeleteObjectVersion*",
+          "s3:PutLifecycleConfiguration*",
+        ],
+        resources: [`${this.bucket.bucketArn}`, `${this.bucket.bucketArn}/*`],
+      });
+    }
+    // The bucket name needs to be passed through an environment variable since
+    // it may not be resolved until deployment time.
+    host.addEnvironment(this.envName(), this.bucket.bucketName);
+
+    super._bind(host, ops);
+  }
+
+  /** @internal */
+  public _toInflight(): core.Code {
+    return core.InflightClient.for(
+      __dirname.replace("awscdk", "tf-aws"),
+      __filename,
+      "BucketClient",
+      [`process.env["${this.envName()}"]`]
+    );
+  }
+
+  private envName(): string {
+    return `BUCKET_NAME_${this.node.addr.slice(-8)}`;
+  }
+}
+
+Bucket._annotateInflight("put", {});
+Bucket._annotateInflight("get", {});
+Bucket._annotateInflight("delete", {});
+Bucket._annotateInflight("list", {});
+Bucket._annotateInflight("put_json", {});
+Bucket._annotateInflight("get_json", {});
```

Comment:
Let's add support for public_url
```suggestion
Bucket._annotateInflight("get_json", {});
Bucket._annotateInflight("public_url", {});
```